components/script/script_thread.rs

```diff
@@ -3817,6 +3842,17 @@ impl ScriptThread {
         };
     }
 
+    fn handle_network_state(&self, is_online: bool, can_gc: CanGc) -> bool {
+        let previous = *self.is_online.lock().unwrap();
+        *self.is_online.lock().unwrap() = is_online;
```

Comment:
You lock .lock().unwrap() twice on the same Mutex, which is inefficient and error-prone:

```
fn handle_network_state(&self, is_online: bool, can_gc: CanGc) -> bool {
    let mut online_lock = self.is_online.lock().unwrap();
    let previous = *online_lock;
    *online_lock = is_online;
    drop(online_lock); // Optional, to release early ?

    if previous != is_online {
        self.fire_network_events(is_online, can_gc);
    }

    false
}
```